runtime/caml/config.h

```diff
@@ -240,7 +240,7 @@ typedef uint64_t uintnat;
 /* Default speed setting for the major GC.  The heap will grow until
    the dead objects and the free list represent this percentage of the
    total size of live objects. */
-#define Percent_free_def 80
+#define Percent_free_def 100
```

Comment:
Just as a data point, in the systems I have tested, 120 has been notably faster without a measurably significant increase in memory usage.

Reply:
To concur, Coq sets 200 with the best-fit, and when I asked they seemed happy with it, so 120 is not shocking.

Reply:
At 200, memory use can increase significantly, but 120 seems like a good trade-off.

According to some measurements of the two (see https://blog.janestreet.com/memory-allocator-showdown/), best-fit at 100 has similar performance to next-fit at 80 while using less memory. This means that the gains from switching to best-fit are entirely allocated to reducing memory use, while at 120 the gains are split between reducing memory use and reducing time spent.

Reply:
(Personally I think that the setting of 200 for Coq was a bit too aggressive. It is a very memory-hungry program, and while everyone is able to wait 10% longer to get their proof to complete, your machine may not necessarily be able to give 10% more memory to the process. I think this hard-limit behavior of memory would suggest a more conservative default, but I guess heavy Coq-users don't have the same perspective as they bought large-memory machines anyway.)

Reply:
I agree with @gasche: let's have default values that don't waste too much memory, because running out of memory and starting to swap is very costly indeed.